src/nvim/window.c

```diff
@@ -1027,9 +1027,11 @@ void ui_ext_win_position(win_T *wp, bool validate)
   }
 }
 
-void ui_ext_win_viewport(win_T *wp)
+void ui_ext_win_viewport(win_T *wp, bool allow_updated)
```

Comment:
would it make senes to use the `validate` pattern found in many other display functions (including `win_ui_flush` which calls this)? Or are the semantics not similar enough?

```suggestion
void ui_ext_win_viewport(win_T *wp, bool validate)
```